src/app/search/search.component.ts

```diff
@@ -180,7 +180,7 @@ export class SearchComponent implements OnInit, OnDestroy {
     if (!paramMap) {
       return;
     }
-    const newAdvancedSearchObject: AdvancedSearchObject = initialAdvancedSearchObject;
+    const newAdvancedSearchObject: AdvancedSearchObject = { ...initialAdvancedSearchObject };
```

Comment:
Tempted to say this should have a comment, given all the time you spent on it.

Reply:
Done.